src/main/java/org/ohdsi/webapi/security/model/EntityPermissionSchema.java

```diff
@@ -2,19 +2,27 @@
 
 import org.ohdsi.webapi.model.CommonEntity;
 import org.ohdsi.webapi.shiro.Entities.RoleEntity;
+import org.ohdsi.webapi.shiro.filters.UpdateAccessTokenFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.ohdsi.webapi.shiro.PermissionManager;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 public abstract class EntityPermissionSchema {
```

Comment:
As we discussed, we are seeking to make the Team feature an extension to WebAPI that folks can opt for and configure. The core developers do not want the Team functionality written into the base WebAPI code (e.g., this class). We agree that the onInsert and onDelete methods will need a bit more logic than in you current method to allow for proper maintainance of shared artifact owner/reader permissions. Your current config-based approach decides on the concrete implementation in this abstract class for that functionality. An advantage of that is that it should apply generally to all the other artifacts through abstract class inheritance. It works for the Team permission management. That might be the right way to go for managing the shared artifact owner/reader permissions too. I write this because I think we can find a general way to assign those permissions that does not need to consider the specific artifact types. I looked at the other Java approaches - Factory, default methods, component instantiation, and optional methods. Factory would require allot of redesign and the other methods do not offer a nice general solution. So, I think this approach is best, possibly with some modifications to abstract out permission management.

Reply:
> I think this approach is best, possibly with some modifications to abstract out permission management.

Thanks. I'll keep it as is for now, and consider some further refinement in a follow-up PR, as I think I'll have to further change this part anyway soon.